c7n/resources/s3.py

```diff
@@ -1465,26 +1465,22 @@ class SetPublicBlock(BucketActionBase):
                 resource: s3
                 filters:
                   - type: check-public-block
-                    scope: BlockPublicAcls
+                    scope: Any
                     enabled: False
                 actions:
                   - type: set-public-block
-                    # scope: <------ optional (All by default)
-                    #   - BlockPublicAcls
-                    #   - IgnorePublicAcls
-                    # state: enable <------ optional (enable by default)
+                  # BlockPublicAcls: true <-- optional, if none provided sets all to true by default
     """
 
     schema = type_schema(
         'set-public-block',
-        scope={
-            'type': 'array',
-            'items': {'type': 'string',
-                     'enum': ['BlockPublicAcls', 'IgnorePublicAcls',
-                        'BlockPublicPolicy', 'RestrictPublicBuckets', 'All']}},
-        state={'type': 'string',
-               'enum': ['enable', 'disable']})
+        BlockPublicAcls={'type': 'boolean'},
```

Comment:
We should allow a `state` setting as well so we can turn all settings off easily with:
```
  actions:
    - type: set-public-block
      state: false
````
I was also going to suggest allowing something like:
```
  actions:
    - type: set-public-block
      state: true
      RestrictPublicBuckets: false
```
and that would set all settings to true except for `RestrictPublicBuckets`.  But that seems slightly contradictory to my previous request to only set a single value and preserve the other's existing values.  I suppose if a setting is specified and `state` is not, you could use existing values, otherwise, `state` ends up being the default value for the other settings?
```
  - type: set-public-block
    BlockPublicAcls: false
```
would only set `BlockPublicAcls` to `false` and preserving others...

Maybe if instead of `state` we call it `All` ... that would imply all settings get that value ... and others could get overridden?
```
  - type: set-public-block
    All: true
    BlockPublicAcls: false
```
hm... not sure I particularly care for that either ... maybe instead of booleans, they should all support 3 values: true, false, or inherit/preserve/existing (pick your favorite)... and then each of the settings explicitly state how the setting is defined.
```
  - type: set-public-block
    state: inherit
    BlockPublicAcls: false
```
and maybe `state` defaults to `true` and folks can explicitly set it to `inherit` the existing values.

Reply:
One nice side affect is that we'd only have to hit the api to get the current settings if someone actually sets `state` to `inherit`.

Reply:
Hmm, interesting. I had similar thoughts of using an inherit flag, definitely makes things more explicit. Let me see how it could be implemented.

Reply:
i'd rather skip inherit, and just have folks be explicit on which attributes they want set, and otherwise leave them as is. that includes getting rid of `All`.  the example policy can show setting setting all the fields on.

incidentally that would then match the behavior of the account action afaics

Reply:
This feels right as a path forward for me too. I like the implied inherit part. All seemed convenient at the start of this but it seems to be adding unnecessary complexity as well. Will update a bit later today when I have a moment. @tjstansell do you feel good about this approach?

Reply:
given that exact consistency was part of the original request from @tjstansell I think its fine to go forward

Reply:
yes it is. sorry ... i swear i thought i'd replied to this already.

Reply:
Sounds good! I have updated to reflect the removal of the all/any stuff. It's got the consistently now alongside the other things we discussed. I admit, I thought I was getting fancy with the All/Any stuff but being explicit just plain works and is much easier to follow along with.

Reply:
While I agree that getting rid of the `inherit` bits makes sense, I think we should still support `state: False` to effectively disable all the settings.  This also keeps consistency with the account action:
```
  actions:
    - type: set-public-block
      state: false
```

Actually, I'm not sure why we don't just cut/paste all the code from the account resource over here.  The logic should be nearly identical, IMO.

Reply:
Hey Todd, currently the action already has an implied truthiness/turn it on to it.

Are you looking for a way so that this 
`
  actions:
    - type: set-public-block
      BlockPublicAcls: false
      IgnorePublicAcls: false
      BlockPublicPolicy: false
      RestrictPublicBuckets: false
`
could be this?
`
  actions:
    - type: set-public-block
       state: false
`
There could be a path forward by getting rid of the implied true so that
`
  actions:
    - type: set-public-block
`
becomes 
`
  actions:
    - type: set-public-block
       state: true
`

In regards to the account filter, could you test to confirm if the state: true works? Looking at the code and testing I was unable to get it to run. I think it can only set all to false in the current state and may need to submit an issue for resolving the bug.